internal/controllers/topology/cluster/conditions.go

```diff
@@ -58,6 +58,22 @@ func (r *Reconciler) reconcileTopologyReconciledCondition(s *scope.Scope, cluste
 		return nil
 	}
 
+	// If the ClusterClass `metadata.Generation` doesn't match the `status.ObservedGeneration` requeue as the ClusterClass
```

Comment:
Cannot select the lines here but, can you please update the comment on the `reconcileTopologyReconciledCondition` to also reflect this new case when the `TopologyReconciled` condition will be set to false.

Also thinking if it would be better to simplify the comment a little to make sure that we dont have to worry about keeping the comment and the implementation in sync. Up to you.